app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -124,19 +125,23 @@ class SearchActivity : AppCompatActivity() {
         }
     }
 
-    private fun searchITunesApi(editText: String) {
-        itunesService.search(editText).enqueue(object : Callback<TrackResponse>{
+    private fun searchITunesApi(query: String) {
+        itunesService.search(query).enqueue(object : Callback<TrackResponse>{
             override fun onResponse(call: Call<TrackResponse>, response: Response<TrackResponse>) {
-                if (response.body()?.results?.isNotEmpty() == true) {
+
+                val trackResponseResult = response.body()?.results
+
+                if (trackResponseResult?.isNotEmpty() == true) {
                     recyclerView.visibility = View.VISIBLE
                     trackList.clear()
-                    trackList.addAll(response.body()?.results!!)
+                    trackList.addAll(trackResponseResult)
                     trackAdapter.notifyDataSetChanged()
                 }
                 else {
                     trackList.clear()
                     trackAdapter.notifyDataSetChanged()
                     recyclerView.visibility = View.GONE
+                    refreshButton.visibility = View.GONE
```

Comment:
Есть удобные экстеншены `isVisible`:
```suggestion
                    refreshButton.isVisible = false
```
Или например еще один:
```suggestion
                    refreshButton.gone()
```